povertymapping/osm.py

```diff
@@ -0,0 +1,115 @@
+import os
+import shutil
+from pathlib import Path
+from zipfile import ZipFile
+
+import geopandas as gpd
+from geowrangler import distance_zonal_stats as dzs
+from geowrangler import vector_zonal_stats as vzs
+from geowrangler.datasets import geofabrik
+from loguru import logger
+
+DEFAULT_POI_TYPES = ["restaurant", "school", "bank", "supermarket", "mall", "atm"]
+
+
+def add_osm_features(
+    aoi,
+    country,
+    cache_dir,
+    poi_types=DEFAULT_POI_TYPES,
+    metric_crs="epsg:3123",
+    inplace=False,
+    force_overwrite=False,
+):
+    """Generates features for the AOI based on OSM data (POIs, roads, etc)."""
+
+    # Load-in the OSM POIs data
+    osm_cache_dir = download_osm_country_data(
+        country, cache_dir, force_overwrite=force_overwrite
+    )
+    osm_pois_filepath = os.path.join(osm_cache_dir, "gis_osm_pois_free_1.shp")
+    osm = gpd.read_file(osm_pois_filepath)
+
+    # Create a copy of the AOI gdf if not inplace to avoid modifying the original gdf
+    if not inplace:
+        aoi = aoi.copy()
+
+    # GeoWrangler: Count number of all POIs per tile
+    aoi = vzs.create_zonal_stats(
+        aoi,
+        osm,
+        overlap_method="intersects",
+        aggregations=[{"func": "count", "output": "poi_count", "fillna": True}],
+    )
+
+    # Count specific aoi types
+    for poi_type in poi_types:
+        # GeoWrangler: Count with vector zonal stats
+        aoi = vzs.create_zonal_stats(
+            aoi,
+            osm[osm["fclass"] == poi_type],
+            overlap_method="intersects",
+            aggregations=[
+                {"func": "count", "output": f"{poi_type}_count", "fillna": True}
+            ],
+        )
+
+        # GeoWrangler: Distance with distance zonal stats
+        col_name = f"{poi_type}_nearest"
+        aoi = dzs.create_distance_zonal_stats(
+            aoi.to_crs(metric_crs),
+            osm[osm["fclass"] == poi_type].to_crs(metric_crs),
+            max_distance=10_000,
+            aggregations=[],
+            distance_col=col_name,
+        ).to_crs("epsg:4326")
+
+        # If no POI was found within the distance limit, set the distance to a really high value
+        aoi[col_name] = aoi[col_name].fillna(value=999999)
```

Comment:
Thanks for flagging JC. Yeah, this was quite arbitrary. 

Will impute with `max_distance`; I think that's a sensible default. Prefer that over leaving as NaN to avoid extra steps on the user's end.